src/main/java/org/codehaus/mojo/license/api/DefaultThirdPartyHelper.java

```diff
@@ -185,9 +195,12 @@ public LicenseMap createLicenseMap( SortedMap<String, MavenProject> dependencies
 
         LicenseMap licenseMap = new LicenseMap();
 
-        for ( MavenProject project : dependencies.values() )
+        synchronized ( dependencies )
```

Comment:
(IIUC) this `dependencies` map is also used in a number of other places, and I don't see the code littered with `synchronized` blocks. Is this location special, or am I missing something?

NB: I'm not suggesting to "synchronize all the things": if that turns out to be necessary, perhaps we should look into a larger redesign. (But given the inactivity on this project, that seems risky, too.)

Reply:
This comes from here 

https://github.com/mojohaus/license-maven-plugin/blob/e29c4bf5c2d31adfd48dbaf0f18f27cee1960285/src/main/java/org/codehaus/mojo/license/AbstractAddThirdPartyMojo.java#L750-L752

in the `AggregatorAddThirdPartyMojo` the `dependencies` is the `artifactCache`, so the iteration must be synchronized.

Reply:
Actually you are right, I have missed one usage here

https://github.com/mojohaus/license-maven-plugin/blob/4b868feaff88186d29179b9077ecca0575ee89a4/src/main/java/org/codehaus/mojo/license/api/DefaultThirdPartyHelper.java#L250